Stripe/STPPaymentContext.m

```diff
@@ -443,6 +443,69 @@ - (void)requestPayment {
 
+            STPSource *source = (STPSource *)self.selectedPaymentMethod;
+
+            switch (source.flow) {
+                case STPSourceFlowNone: {
+                    // TODO: if this is a card, check if 3DS should be used
+                    STPPaymentResult *result = [[STPPaymentResult alloc] initWithSource:source];
+                    [self.delegate paymentContext:self didCreatePaymentResult:result completion:^(NSError * _Nullable error) {
+                        stpDispatchToMainThreadIfNecessary(^{
+                            if (error) {
+                                [self.delegate paymentContext:self didFinishWithStatus:STPPaymentStatusError error:error];
+                            } else {
+                                [self.delegate paymentContext:self didFinishWithStatus:STPPaymentStatusSuccess error:nil];
+                            }
+                        });
+                    }];
+
+                }
+                    break;
+                case STPSourceFlowRedirect: {
+                    self.configuration.sourceURLRedirectBlock(self.apiClient, source, ^(STPSource *finishedSource, NSError *error) {
+                        stpDispatchToMainThreadIfNecessary(^{
+                            if (error) {
+                                [self.delegate paymentContext:self didFinishWithStatus:STPPaymentStatusError error:error];
+                            } else {
+                                switch (finishedSource.status) {
+                                    case STPSourceStatusChargeable:
+                                    case STPSourceStatusConsumed:
+                                        [self.delegate paymentContext:self
+                                                  didFinishWithStatus:STPPaymentStatusSuccess
+                                                                error:nil];
+                                        break;
+                                    case STPSourceStatusPending:
+                                        [self.delegate paymentContext:self
+                                                  didFinishWithStatus:STPPaymentStatusPending
+                                                                error:nil];
+                                        break;
+                                    case STPSourceStatusCanceled:
+                                    case STPSourceStatusFailed:
+                                    case STPSourceStatusUnknown:
+                                        // TODO: should this be user cancelled in the failed state?
+                                        // (which means the user did not do the required action during the redirect)
+                                        [self.delegate paymentContext:self
+                                                  didFinishWithStatus:STPPaymentStatusError
+                                                                error:nil];
```

Comment:
Users will be expecting a non-nil `error` here – can we add some new NSError objects for here and below?